lib/scheduler.js

```diff
@@ -184,6 +187,31 @@ class Scheduler extends EventEmitter {
       await this.connection.redis.zrem(this.connection.key('delayed_queue_schedule'), timestamp)
     }
   }
+
+  async checkStuckWorkers () {
+    if (!this.options.stuckWorkerTimeout) { return }
+
+    const keys = await this.connection.redis.keys(this.connection.key('worker', 'ping', '*'))
```

Comment:
what's nice about using this new namespace, is that if Workers in other languages don't set this key, we won't mess with them

Reply:
I'm just wondering whether it wouldn't be more efficient to use a sorted set and ZADD (for setting heartbeats) + ZRANGEBYSCORE (for getting possibly-stale workers) + ZREMRANGEBYSCORE (for removing said workers) + ZREM (for clearing an individual worker on shutdown), with the score being the timestamp of the last worker heartbeat, instead of first getting the keys, then getting the last heartbeat for each worker key, then deciding what to remove.

Actually I'd have to benchmark this, hmm, as it has to weigh the sorted sets' underlying skip lists vs what I believe is the linear scan performed by KEYS.

I'd like to prototype that but I'm not sure whether I should do it against this PR or wait until it gets merged (since it's already accepted).